services/wireguard/service.go

```diff
@@ -63,11 +63,11 @@ type Config struct {
 	Consumer network.Consumer
 }
 
-// Network represents Wireguard network instance, it provide information
-// required for establishing connection between service provider and consumer.
-type Network interface {
+// Provider represents Wireguard network instance that serves consumers.
+// It returns information required for establishing connection between service provider and consumer.
+type Provider interface {
```

Comment:
Provider interface which has method Provider which returns network.Provider :(

Reply:
It was completely refactored.